autobot/telegram/parser.py

```diff
@@ -697,3 +694,417 @@ def _parse_sticker (self,key:str,val:dict) -> Sticker|StickerSet:
+        elif (key == "poll_option"):
+            poll_obj = PollOption()
+            for k,v in val.items():
+                setattr(poll_obj,k,v)
+        return poll_obj
+
+    def _parse_polloption(self,key:str,val:dict) -> PollOption:
+        polloption_obj = PollOption()
+        for k,v in val.items():
+            setattr(polloption_obj,k,v)
+        return polloption_obj
+
+    def _parse_document(self,key:str,val:dict) -> Document:
+        document_obj = Document()
+        for k,v in val.items():
+            match k:
+                case "thumb":
+                    thumb = self._parse_photosize(k,v)
+                    setattr(document_obj,k,thumb)
+                case _:
+                    setattr(document_obj,k,v)
+        return document_obj
+
+    def _parse_webapp(self,key:str,val:dict) -> WebAppData:
+        webapp_obj = WebAppData()
+        for k,v in val.items():
+            setattr(webapp_obj,k,v)
+        return webapp_obj
+
+    def _parse_callbackquery(self,key:str,val:dict) -> CallBackQuery:
+        callbackquery_obj = CallBackQuery()
+        for k,v in val.items():
+            setattr(callbackquery_obj,k,v)
+        return callbackquery_obj
+
+    def _parse_inputmessagecontent(self,key:str,val:dict) -> InputMessageContent:
+        inputmessagecontent_obj = None
+        if (key == "input_text_message_content"):
+            inputmessagecontent_obj = InputTextMessageContent()
+            for k,v in val.items():
+                setattr(inputmessagecontent_obj,k,v)
+        elif (key == "input_location_message_content"):
+            inputmessagecontent_obj = InputLocationMessageContent()
+            for k,v in val.items():
+                setattr(inputmessagecontent_obj,k,v)
+        elif (key == "input_venue_message_content"):
+            inputmessagecontent_obj = InputVenueMessageContent()
+            for k,v in val.items():
+                setattr(inputmessagecontent_obj,k,v)
+        return inputmessagecontent_obj
+
+    def _parse_inlinekeyboardbuttons(self,key:str,val:dict) -> InlineKeyboardButton:
+        inlinekeyboardbutton_obj = InlineKeyboardButton()
+        for k,v in val.items():
+            match k:
+                case "text"|"url"|"callback_data"|"switch_inline_query"|"switch_inline_query_current_chat"|"callback_game"|"pay":
+                    setattr(inlinekeyboardbutton_obj,k,v)
+                case _:
+                    setattr(inlinekeyboardbutton_obj,k,v)
+        return inlinekeyboardbutton_obj
+
+
+    def _parse_stickerset(self,key:str,val:dict) -> StickerSet:
+        stickerset_obj = StickerSet()
+        for k,v in val.items():
+            match k:
+                case "stickers":
+                    stickers = []
+                    for sticker in v:
+                        stickers.append(self._parse_stickers(k,sticker))
+                    setattr(stickerset_obj,k,stickers)
+                case _:
+                    setattr(stickerset_obj,k,v)
```

Comment:
There's no need for creating a list and appending it because the dictionary we are parsing already does that for us more efficiently.

so there's no need for this `match case` statement, just use the `for` loop to set attribute. I remember mentioning that we shouldn't manipulate the data the we receive, the job of the parser is just to direct construction of the object tree without interfering with the data we receive

Reply:
@kallyas I've seen what you're doing now. Sorry I didn't see it earlier. I think this is alright then

Reply:
@kallyas did you make changes to this? If you did, I request that you revert them. Your solution was alright